test/base-tests/run.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+set -euox pipefail
+
+CHECKOUT_DIRECTORY="${PWD}"
+WORKING_DIRECTORY=/usr/libexec/osbuild-composer
+TESTS_PATH=/usr/libexec/tests/osbuild-composer
+
+PASSED_TESTS=()
+FAILED_TESTS=()
+
+TEST_RUNNER=/usr/libexec/tests/osbuild-composer/osbuild-base-test-runner
+
+function retry {
+    local count=0
+    local retries=5
+    until "$@"; do
+        exit=$?
+        count=$(($count + 1))
+        if [[ $count -lt $retries ]]; then
+            echo "Retrying command..."
+            sleep 1
+        else
+            echo "Command failed after ${retries} retries. Giving up."
+            return $exit
+        fi
+    done
+    return 0
+}
+
+# Get OS details.
+source /etc/os-release
+
+# Register RHEL if we are provided with a registration script.
+if [[ -n "${RHN_REGISTRATION_SCRIPT:-}" ]] && ! sudo subscription-manager status; then
+    sudo chmod +x $RHN_REGISTRATION_SCRIPT
+    sudo $RHN_REGISTRATION_SCRIPT
+fi
+
+# Restart systemd to work around some Fedora issues in cloud images.
+sudo systemctl restart systemd-journald
+
+# Remove Fedora's modular repositories to speed up dnf.
+sudo rm -f /etc/yum.repos.d/fedora*modular*
+
+# Enable fastestmirror and disable weak dependency installation to speed up
+# dnf operations.
+echo -e "fastestmirror=1\ninstall_weak_deps=0" | sudo tee -a /etc/dnf/dnf.conf
+
+# Ensure we are using the latest dnf since early revisions of Fedora 31 had
+# some dnf repo priority bugs like BZ 1733582.
+# NOTE(mhayden): We can exclude kernel updates here to save time with dracut
+# and module updates. The system will not be rebooted in CI anyway, so a
+# kernel update is not needed.
+if [[ $ID == fedora ]]; then
+    sudo dnf -y upgrade --exclude kernel --exclude kernel-core
+fi
+
+# Add osbuild team ssh keys.
+cat schutzbot/team_ssh_keys.txt | tee -a ~/.ssh/authorized_keys > /dev/null
+
+# Set up a dnf repository for the RPMs we built via mock.
+sudo cp osbuild-mock.repo /etc/yum.repos.d/osbuild-mock.repo
+sudo dnf repository-packages osbuild-mock list
+
+# Ensure osbuild-composer-tests is installed.
+retry sudo dnf -y install osbuild-composer-tests
+
+ls -l /usr/libexec/tests/osbuild-composer
+
+# Change to the working directory.
+cd $WORKING_DIRECTORY
+
+# Run
+sudo --preserve-env $TEST_RUNNER /usr/libexec/tests/osbuild-composer/osbuild-weldr-tests
```

Comment:
I think you want this in quotes as well, but depends on what this is exactly used for.